File: src/main/kotlin/com/feakin/intellij/FeakinLanguage.java
package com.feakin.intellij;

import com.intellij.lang.Language;

public class FeakinLanguage extends Language {

    public static final FeakinLanguage INSTANCE = new FeakinLanguage();

    public static FeakinLanguage getInstance() {
        return INSTANCE;
    }

    private FeakinLanguage() {
        super("Feakin");
    }

}

File: src/main/kotlin/com/feakin/intellij/FeakinLanguage.kt
package com.feakin.intellij

import com.intellij.lang.Language

object FeakinLanguage : Language("Feakin", "text/feakin", "text/x-feakin", "application/x-feakin") {
    override fun isCaseSensitive() = true

    override fun getDisplayName() = "Feakin"
}